ibft/eventqueue/queue.go

```diff
@@ -1,36 +1,54 @@
 package eventqueue
 
-import "sync"
+import (
+	"github.com/bloxapp/ssv/utils/logex"
+	"go.uber.org/zap"
+	"sync"
+	"time"
+)
 
-// Queue is a simple struct that manages a queue of functions. Thread safe
-type Queue struct {
+// Event represent some function
+type Event func()
+
+// EventQueue is the interface for managing a queue of functions
+type EventQueue interface {
+	Add(Event) bool
+	Pop() Event
+	ClearAndStop()
+}
+
+// queue thread safe implementation of EventQueue
+type queue struct {
 	stop  bool
-	queue []func()
+	queue []Event
 	lock  sync.Mutex
 }
 
-// New returns a new instance of Queue
-func New() *Queue {
-	return &Queue{
-		queue: make([]func(), 0),
+// New returns a new instance of queue
+func New() EventQueue {
+	q := queue{
+		queue: make([]Event, 0),
 		lock:  sync.Mutex{},
 	}
+	return &q
 }
 
 // Add will add an an item to the queue, thread safe.
-func (q *Queue) Add(f func()) {
+func (q *queue) Add(e Event) bool {
```

Comment:
@nivBlox why return bool from add?

Reply:
@alonmuroch to identify cases where the event wasn't added to the queue (if it was stopped)